extensions-core/druid-catalog/src/main/java/org/apache/druid/catalog/http/CatalogResource.java

```diff
@@ -0,0 +1,747 @@
+  )
+  {
+    Response response = authorizeTable(tableId, newSpec, req);
+    if (response != null) {
+      return response;
+    }
+    try {
+      long newVersion = catalog.tables().updatePayload(tableId, action);
+      return Actions.okWithVersion(newVersion);
+    }
+    catch (NotFoundException e) {
+      return Response.status(Response.Status.NOT_FOUND).build();
+    }
+    catch (Exception e) {
+      return Actions.exception(e);
+    }
+  }
+
+  /**
+   * Move a single column to the start end of the column list, or before or after
+   * another column. Both columns must exist. Returns the version of the table
+   * after the update.
+   * <p>
+   * The operation is done atomically so no optimistic locking is required.
+   *
+   * @param dbSchema
+   * @param name
+   * @param command
+   * @param req
+   * @return
+   */
+  @POST
+  @Path("/tables/{dbSchema}/{name}/moveColumn")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response moveColumn(
+      @PathParam("dbSchema") final String dbSchema,
+      @PathParam("name") final String name,
+      final MoveColumn command,
+      @Context final HttpServletRequest req
+  )
+  {
+    if (command == null) {
+      return Actions.badRequest(Actions.INVALID, "A MoveColumn object is required");
+    }
+    if (Strings.isNullOrEmpty(command.column)) {
+      return Actions.badRequest(Actions.INVALID, "A column name is required");
+    }
+    if (command.where == null) {
+      return Actions.badRequest(Actions.INVALID, "A target location is required");
+    }
+    if ((command.where == Position.BEFORE || command.where == Position.AFTER) && Strings.isNullOrEmpty(command.anchor)) {
+      return Actions.badRequest(Actions.INVALID, "A anchor column is required for BEFORE or AFTER");
+    }
+    return incrementalUpdate(
+        TableId.of(dbSchema, name),
+        null,
+        req,
+        (spec) -> spec.withColumns(command.perform(spec.columns()))
+    );
+  }
+
+  /**
+   * Hide or unhide columns. If both appear, hide takes precedence. Returns the
+   * new table version.
+   */
+  @POST
+  @Path("/tables/{dbSchema}/{name}/hideColumns")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response hideColumns(
+      @PathParam("dbSchema") final String dbSchema,
+      @PathParam("name") final String name,
+      final HideColumns command,
+      @Context final HttpServletRequest req
+  )
+  {
+    return incrementalUpdate(
+        TableId.of(dbSchema, name),
+        null,
+        req,
+        (spec) -> {
+          if (!AbstractDatasourceDefn.isDatasource(spec.type())) {
+            throw new ISE("hideColumns is supported only for data source specs");
+          }
+          @SuppressWarnings("unchecked")
+          List<String> hiddenProps = (List<String>) spec.properties().get(AbstractDatasourceDefn.HIDDEN_COLUMNS_PROPERTY);
+          return spec.withProperty(
```

Comment:
Does it make a difference if the hidden columns are or are not also listed as columns in table column list itself?

Reply:
makes sense